std/assembly/typedarray.ts

```diff
@@ -1260,3 +1304,34 @@ function REVERSE<TArray extends ArrayBufferView, T>(array: TArray): TArray {
   }
   return array;
 }
+
+// @ts-ignore: decorator
+@inline
+function WRAP<TArray extends ArrayBufferView, T>(buffer: ArrayBuffer, byteOffset: i32 = 0, length: i32 = -1): TArray {
+  if (byteOffset < 0 || byteOffset >= buffer.byteLength) {
+    throw new RangeError("Start offset out of bound");
+  }
+  var byteLength: i32;
+  if (length < 0) {
+    if (length == -1) {
+      let size = sizeof<T>();
+      if (buffer.byteLength % size == 0) {
```

Comment:
Maybe, to avoid conversion to f64 though fmod:

```ts
const mask = <i32>((1 << alignof<T>()) - 1);
if (cachedBufferByteLength & mask) {
  // error
} else {
  // good
}
```

Reply:
Fixed